### ISSUE TITLE
no-unused-expressions on function call using ?. operator

### ISSUE THREAD

**Issue by vinodsai-a** [eslint/eslint#13637]
**Tell us about your environment**

Node version: v12.18.3
npm version: v6.14.6
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
   onChange?.(color);
```

```bash

```

**What did you expect to happen?**
No error when calling a function using **?.** operator (ex: function?.())

**What actually happened? Please include the actual, raw output from ESLint.**
Shows Error: Expected an assignment or function call and instead saw an expression.eslintno-unused-expressions

**Are you willing to submit a pull request to fix this bug?**

**Comment by anikethsaha** [comment]
Thanks for the issue.

IMO, `ChainExpression` (`onChange?.(color);`) should not report only when `allowShortCircuit` is `true` like the `LogicalAssignment` (`onChange && onChange (color);`)

Something like this [here](https://github.com/eslint/eslint/blob/master/lib/rules/no-unused-expressions.js#L131-L133)

```js
ChainExpression(node) {
         if (allowShortCircuit) {
                    return Checker.isDisallowed(node.expression);
          }
         return true;
}
```

Cause there is an inconsistency i.e [this example](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLWV4cHJlc3Npb25zOiBbXCJlcnJvclwiXSovXG5cbm9uQ2hhbmdlPy4oY29sb3IpXG5vbkNoYW5nZSAmJiBvbkNoYW5nZShjb2xvcik7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Im5vLXVudXNlZC12YXJzIjoyfSwiZW52Ijp7fX19) here, `onChange?.(color)` should be reported. 


Though I am not sure about this, cc @eslint/eslint-team thoughts ?

**Comment by mdjermanovic** [comment]
`onChange?.(color);` is a valid code, it doesn't look like a possible error and shouldn't be reported by this rule.

`onChange && onChange (color);` could be a pattern where `&&` is used instead of `if`, but could be also a bug in the code. That's why this rule has the `allowShortCircuit` option: users who are not using this pattern should set `allowShortCircuit: false` (default) so the rule can report possible errors in these cases.

@vinodsai-a this rule supports optional chaining since [v7.5.0](https://eslint.org/blog/2020/07/eslint-v7.5.0-released). Can you upgrade ESLint to the latest version?

**Comment by nzakas** [comment]
@anikethsaha please only mention the whole team as a last resort. Doing so trigger notifications for more than ten people, and we don’t want to flood everyone’s notifications unless it’s absolutely necessary.

**Comment by anikethsaha** [comment]
> @anikethsaha please only mention the whole team as a last resort. Doing so trigger notifications for more than ten people, and we don’t want to flood everyone’s notifications unless it’s absolutely necessary.

My bad, Yea Sure, I will note this.

**Comment by vinodsai-a** [comment]
> `onChange?.(color);` is a valid code, it doesn't look like a possible error and shouldn't be reported by this rule.
> 
> `onChange && onChange (color);` could be a pattern where `&&` is used instead of `if`, but could be also a bug in the code. That's why this rule has the `allowShortCircuit` option: users who are not using this pattern should set `allowShortCircuit: false` (default) so the rule can report possible errors in these cases.
> 
> @vinodsai-a this rule supports optional chaining since [v7.5.0](https://eslint.org/blog/2020/07/eslint-v7.5.0-released). Can you upgrade ESLint to the latest version?

Thanks, @mdjermanovic. It worked after upgrade.
